### PR TITLE
70 GitHub revoked

### LINKED ISSUE #70: Handle the case where github auth has been revoked
### How to reproduce:
- Make sure you currently have github oauth access within ployst.
- Go to https://github.com/settings/applications
- Go "Authorized Applications".
- Revoke the permission of the relevant Deploystream app.
- Attempt to view http://localhost:5000/ui/#/projects
### Expected behaviour
- Github asks for your permission for this app again
### Actual behaviour
- Silent 401s and 500s in logs and the user sees "Loading..." for their organizations.

### PR DESCRIPTION
**Pull request by alexcouper**
Resolves #70

### CODE REVIEW

File: ployst/github/urls.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
     # authentication to github account
     url(r'^oauth-start/', oauth.start, name='oauth-start'),
     url(r'^oauth-confirmed/', oauth.receive, name='oauth-callback'),
+    url(r'^oauth-access-token', oauth.token, name='oauth-token'),
```
Comment (txels) [inline ployst/github/urls.py:14]: to backslash or to not backslash...

File: ployst/github/views/oauth.py [hunk 2]
```diff
@@ -88,3 +91,22 @@ def exchange_for_access_token(user_id, code_to_exchange):
...
+    Get and validate the oauth token for github associated with this account.
+    """
+    token = client.get_access_token(request.user.id, 'github')
+
+    # Validate credentials
+    try:
+        gh_login(token=token['token']).user()
+    except GitHubError:
+        # Return a blank token
+        tokens = []
```
Comment (txels) [inline ployst/github/views/oauth.py]: Should we also delete the token from core?

File: README [hunk 3]
```diff
@@ -1,2 +1,8 @@
 Ployst
 ======
+
+Running tests
+-------------
+
+python manage.py test
+
```
Comment (txels) [inline README]: True. I typically look up in tox.ini what command I need to run JS tests.
It's `karma start [--single-run]`.

Comment (txels) [inline README]: Rename to README.md?

File: ployst/github/views/oauth.py (hunk 2 above)
Comment (txels) [inline ployst/github/views/oauth.py]: (although functionally it won't make a difference)

Comment (alexcouper) [inline ployst/github/views/oauth.py]: Deleted.

**Comment by alexcouper** [comment]
This is ready for another pass.

File: ployst/core/accounts/views.py [hunk 4]
```diff
@@ -94,6 +94,10 @@ class UserTokenViewSet(PermissionsViewSetMixin, ModelViewSet):
     model = UserOAuthToken
     filter_fields = ('user', 'identifier')
 
+    def delete(self, request, *args, **kwargs):
+        self.kwargs = {'pk': request.GET['id']}
+        return self.destroy(request, *args, **kwargs)
```
Comment (txels) [inline ployst/core/accounts/views.py:99]: strange... works for me though.

File: ployst/core/client.py [hunk 5]
```diff
@@ -20,28 +20,34 @@ def __init__(self, base_url, access_token):
...
+    def _http(self, method, path, data=None, params=None):
         meth = getattr(requests, method)
         response = meth(
             '{0}{1}'.format(self.base_url, path),
             data=data,
+            params=params,
             headers={'X-Ployst-Access-Token': self.access_token}
         )
-        if response.status_code not in [200, 201]:
+        if response.status_code > 299:
```
Comment (txels) [inline ployst/core/client.py:31]: technically speaking, only >=400 are errors...

**Pull request merged**